proposals/0540-vision-for-layout-conformance-parity.md

```diff
@@ -0,0 +1,238 @@
+
+### Browser engines
+
+It is tempting to try to reuse an already developed styling engine from a browser. Major advantages are conformance and capabilities, but these are tied to internals of each browser, and add a large footprint and complexity that is undesirable. We would also need to build in Yoga bug compatibility. It would effectively entail rebuilding a new layout engine, heavily forked from a one-time snapshot of a browser engine.
+
+### Taffy
+
+[Taffy](https://github.com/DioxusLabs/taffy) is a Flexbox layout engine written in Rust, authored by Emil Sjölander, a main contributor of Yoga. It is seeing active development under new hands, and has some capabilities that Yoga doesn’t (e.g. it recently added `align-items: 'space-evenly'`). It does not have bug compatibility with the existing RN ecosystem, its rollout/production usage is unknown, and being owned outside of Meta limits our ability to direct the engine in ways which are targeted towards RN’s use-cases.
+
+### Platform-specific controls
+
+Platform-holders like Microsoft and Google have looked at (or have) implemented their own FlexBox controls, such as `google/FlexboxLayout`. For cases where these controls may be made in-box, we get to punt on a lot of complexity, and make React Native apps smaller. Not all platforms have native flexbox controls though, they do not match the set of capabilities we want, and they do not have bug compatibility or Meta ownership.
+
+## Validating conformance
+
+We know about some subset of conformance issues, but there are likely to be more, and as we add new capabilities, we must be systematic in ensuring a correct implementation. Browsermakers share a set of tests for validating conformance called [WPT (Web Platform Tests)](https://github.com/web-platform-tests/wpt/tree/master/css). These have already been leveraged by React Native for pointer events, by porting the HTML to React Native. An alternative to porting the code may be to rely on Yoga’s existing capabilities of generating unit test derived from how Chrome lays out HTML fixtures.
+
+## Specific W3C conformance issues
```

Comment:
You can add `aspect-ratio` to the list here. Yoga's implementation of `aspect-ratio` predates the property being added to CSS (by around 3 years), and there are a number of comments around the Yoga codebase stating that the aspect-ratio property is not spec compliant.